server/src/main/java/org/elasticsearch/action/delete/DeleteRequest.java

```diff
@@ -215,6 +250,13 @@ public void readFrom(StreamInput in) throws IOException {
         }
         version = in.readLong();
         versionType = VersionType.fromValue(in.readByte());
+        if (in.getVersion().onOrAfter(Version.V_7_0_0)) {
+            ifSeqNoMatch = in.readZLong();
```

Comment:
this can be -2 (unassigned). I wonder if we should call the validate method on these once deserialized and under assertion? I think that will give us a much better testing coverage?